src/Cli/dotnet/CommonOptions.cs

```diff
@@ -100,6 +116,57 @@ public static bool VerbosityIsDetailedOrDiagnostic(this VerbosityOptions verbosi
                 verbosity.Equals(VerbosityOptions.d) ||
                 verbosity.Equals(VerbosityOptions.detailed);
         }
+
+        internal static IEnumerable<string> ResolveArchOptionToRuntimeIdentifier(string arg, ParseResult parseResult)
+        {
+            if (parseResult.BothArchAndOsOptionsSpecified())
+            {
+                return Array.Empty<string>();
```

Comment:
I think it's worth adding a comment here that we return empty array since we rely on the `ResolveOsOptionToRuntimeIdentifier` to be called which will handle the case of both specified at the same time.

Reply:
I just read "AndOs" as "Andos" and `ResolveOs` as `Resolves`. When skimming through these things happen. It would be better to expand `OS` to `OperatingSystem` everywhere in the sources especially in the variable names. It might be long but will be readable and more legible.